### PR TITLE
Feature/assignment 3

### PR DESCRIPTION
**Pull request by Sneha1251**
Screenshot from 2024-04-08 15-03-06
Screenshot from 2024-04-08 15-03-10

### CODE REVIEW

File: react-app/src/Assignment-3/Question6/Products.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useParams, useNavigate } from "react-router-dom";
+
+const items = [
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question6/Products.jsx]: create new file constants.js
export const items = [

File: react-app/src/Assignment-3/Question2/Question-2.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// Modify the parent component to provide both the authentication and preferences contexts.
+// Create a child component that consumes both contexts.
+// Allow the user to change their theme preference (light/dark) using a button in the child component.
+// Display the theme preference in the UI and adjust the component's styling accordingly.
+
+import React, { createContext, useState, useContext } from "react";
+import AuthProvider from "../Question1/Question-1";
+const PreferencesContext = createContext();
+
+const PreferencesProvider = () => {
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question2/Question-2.jsx:12]: Use another file for creating context from next time onwards.

File: react-app/src/Assignment-3/Question6/Products.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }}
+      >
+        GO Back
+      </button>
+    </div>
+  );
+};
+
+export default Products;
+export { items };
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question6/Products.jsx]: remove the items from here

File: react-app/src/Assignment-3/Question4/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  };
+  
+  return (
+    <>
+      <div>
+        <input
+          type="text"
+          placeholder="enter name"
+          onChange={(e) => {
+            setname(e.target.value);
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question4/Login.jsx]: follow camelCase for all name, event handler

File: react-app/src/Assignment-3/Question4/Protected.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+const Protected = (props) => {
+  const { Component } = props;
+  const navigate = useNavigate(-1);
+
+  useEffect(() => {
+    let login = localStorage.getItem("login");
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question4/Protected.jsx]: use const

File: react-app/src/Assignment-3/Question4/Question-4.jsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+import React from "react";
+import { Routes, Route } from "react-router-dom";
+import Login from "./Login";
+import Home from "../Question3/Home";
+import About from "../Question4/About";
+import Protected from "../Question4/Protected";
+
+const ProtectedAuthentication=()=>{
+ return(
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question4/Question-4.jsx]: make one file for all routes as currently created for each module seperatly.

File: react-app/src/Assignment-3/Question3/Question-3.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+// create pages: Home and About.
+// Implement routing using React Router DOM.
+// Create navigation links to switch between the Home and About pages.
+// Display appropriate content on each page.
+// Add a "404 Not Found" page for any invalid routes.
+
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question3/Question-3.jsx:6]: add component for 404 page

File: react-app/src/Assignment-3/Question4/Protected.jsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+const Protected = (props) => {
```
Comment (DevGarima) [inline react-app/src/Assignment-3/Question4/Protected.jsx]: use proper name in all places

**Review by DevGarima — CHANGES_REQUESTED** [review]

File: react-app/src/Assignment-3/Question2/Question-2.jsx (hunk 2 above)
Comment (Sneha1251) [reply react-app/src/Assignment-3/Question2/Question-2.jsx:12]: Done

File: react-app/src/Assignment-3/Question3/Question-3.jsx (hunk 7 above)
Comment (Sneha1251) [reply react-app/src/Assignment-3/Question3/Question-3.jsx:6]: Done

File: react-app/src/Assignment-3/Question4/Login.jsx (hunk 4 above)
Comment (Sneha1251) [reply react-app/src/Assignment-3/Question4/Login.jsx]: done

File: react-app/src/Assignment-3/Question6/Products.jsx (hunk 1 above)
Comment (Sneha1251) [reply react-app/src/Assignment-3/Question6/Products.jsx]: Done

**Review by DevGarima — APPROVED** [review]

**Pull request merged**